### ISSUE TITLE
Crash caused by IPC happens after distributing

### ISSUE THREAD

**Issue by sheyifan** [electron/electron#26674]
### Preflight Checklist

* [x] I can successfully run Electron application by `npm install && npm start`. However, after distributing application according to offical documantation, I found that the executable crash. After set codes about IPC to comment and re-distributing, the executable can run successfully. 

### Issue Details

* **Electron Version:**
  * 10.0.0
* **Operating System:**
  * Window 10 (1909)

**Comment by ckerr** [comment]
Thanks for reporting this and helping to make Electron better!

Would it be possible for you to make a standalone testcase with only the code necessary to reproduce the issue? For example, [Electron Fiddle](https://www.electronjs.org/fiddle) is a great tool for making small test cases and makes it easy to publish your test case to a [gist](https://gist.github.com) that Electron maintainers can use.

Stand-alone test cases make fixing issues go more smoothly: it ensure everyone's looking at the same issue, it removes all unnecessary variables from the equation, and it can also provide the basis for automated regression tests.

I'm adding the `blocked/needs-repro` label for this reason. After you make a test case, please link to it in a followup comment.

Thanks in advance! Your help is appreciated.

**Comment by sheyifan** [comment]
I have solved the problem. Thanks for your advice. It helps me a lot.
The crash is caused by incorrect file path resolution after distribution. I used [Electron Forge](https://www.electronforge.io/), which is suggested at [Electron quick start](https://www.electronjs.org/docs/tutorial/quick-start#package-and-distribute-the-application). My application use something like `path.resolve('./data.json')` in `main.js` to get relative path. When `main.js` is at the same directory as `data.json`, everything works well. However after distributing, `data.json` is deployed in `<projectpath>/resources/app/data.json`, but the exucutable is deployed into `<projectpath>`, and the executable cannot find `data.json` by resolved file path. My app will continously walk the directory tree when it cannot find `data.json`, which caused crash to happen.
